practices/appeng/adrs/0005-postgresql-as-relational-store.md

```diff
@@ -0,0 +1,95 @@
+# 0005 - Use PostgreSQL as a relational store
+
+**Status:** In Progress
+
+**Date Accepted:** TBD
+
+**Reviewers:** @reggieriser @mikena-truss @sandy-wright @felipe-lee @ferlatte
+
+## Context
+
+It is typical for our projects to require the persistent storage of data. A relational database is
+often a good choice for this as it allows us to predefine a schema for and relationships
+between data. A database management system (DBMS) can interpret and enforce these rules
+while providing a means to efficiently and safely interact with the data. This ADR documents the
+reasoning behind our recommendation of a preferred DBMS.
+
+## Decision
+
+Use [PostgreSQL](https://www.postgresql.org) when you need a relational store.
+
+## Why is this Applicable to the Practice as a Whole
+
+Since the requirement for a relational data store is common, there are advantages of us standardizing
+on one DBMS across the practice. Benefits include:
+
+- ability to transfer knowledge across projects
+- reduced ramp-up time when switching projects
+- support by cloud providers, allowing reuse of infrasec practices
+- norming on a common SQL/DDL dialect
+
+Historically, most Truss projects deploy a database on AWS. Note that AWS provides two PostgreSQL options:
+RDS PostgreSQL and Aurora PostgreSQL. The typical Truss project will likely use RDS PostgreSQL since Aurora targets
+very large or high-availability sites. That said, evaluate the features and pricing for each against your project's
+needs.
+
+## When to Not Implement This Decision
+
+- If the project's client/sponsor has an established and/or inflexible existing DBMS requirement
+- If the nature of the data lends itself to a different type of data store (NoSQL database, graph database, etc.)
+- The project is so small, self-contained, or temporary that a lightweight solution like SQLite is more appropriate
```

Comment:
Curious if this is relevant. Maybe personal bias, but postgres has become accessible enough to run where the migration from sqlite to postgres outweighs the ease of using sqlite

Reply:
I guess I was just hedging a bit for cases (like an example, starter, or template project) where being able to spin something up with a file-based database like SQLite was helpful in removing a little bit of setup friction.  I don't anticipate that we'd ever use SQLite in a deployed project.